Sources/SentryCrash/Recording/SentryCrashReport.c

```diff
@@ -1207,12 +1205,12 @@ writeBinaryImage(
 static void
 writeBinaryImages(const SentryCrashReportWriter *const writer, const char *const key)
 {
-    const int imageCount = sentrycrashdl_imageCount();
-
+    const int imageCount = sentrycrashbic_imageCount();
     writer->beginArray(writer, key);
     {
         for (int iImg = 0; iImg < imageCount; iImg++) {
-            writeBinaryImage(writer, NULL, iImg);
+            SentryCrashBinaryImage *image = sentrycrashbic_getBinaryImageCache(iImg);
```

Comment:
`h`: With the current approach, we either need a copy of all binary images or a lock on them cause the imageCount could change while we're iterating over the list, so we could get index out of bounds or end up writing an image twice.

Reply:
Its very unlikely to have a new lib being loaded after a crash and locks are not async-signal-safe.
We could add a nil check here because the lib has bounds protection.
I don't think Its possible to add the same library twice because they are added to the end of the array.